scripts/build_test.py

```diff
@@ -14,29 +14,270 @@
+                'path/to/file.min.js', 'fedcba'), 'path/to/file.min.fedcba.js')
+
+    def test_ensure_files_exist(self):
+        """Test _ensure_files_exist raises exception with a non-existent
+        filepath.
+        """
+        random_filepaths = [
+            os.path.join(build.THIRD_PARTY_GENERATED_DEV_DIR, 'random1.js')]
+        with self.assertRaises(OSError) as fileNotExist:
+            build._ensure_files_exist(random_filepaths)
+        # Exception will be raised at first file determined to be non-existent.
+        self.assertTrue(
+            ('File %s does not exist.') % random_filepaths[0] in
+            fileNotExist.exception)
+
+    def test_get_file_count(self):
+        """Test get_file_count to return the correct number of files, excluding
+        file with extensions in FILE_EXTENSIONS_TO_IGNORE.
+        """
+        all_inclusive_file_count = 0
+        for _, _, files in os.walk(
+                build.EXTENSIONS_DIRNAMES_TO_DIRPATHS['dev_dir']):
+            all_inclusive_file_count += len(files)
+        ignored_file_count = 0
+        for _, _, files in os.walk(
+                build.EXTENSIONS_DIRNAMES_TO_DIRPATHS['dev_dir']):
+            for filename in files:
+                if any(filename.endswith(p)
+                       for p in build.FILE_EXTENSIONS_TO_IGNORE):
+                    ignored_file_count += 1
+        self.assertEqual(
+            all_inclusive_file_count - ignored_file_count,
+            build.get_file_count(
+                build.EXTENSIONS_DIRNAMES_TO_DIRPATHS['dev_dir']))
+
+    def test_compare_file_count(self):
+        """Test _compare_file_count to raise exception when there is a
+        mismatched file count between 2 dirs.
+        """
+        with self.assertRaises(ValueError) as incorrectFileCount:
+            build._compare_file_count(
+                build.EXTENSIONS_DIRNAMES_TO_DIRPATHS['dev_dir'],
+                build.TEMPLATES_CORE_DIRNAMES_TO_DIRPATHS['dev_dir'])
+        source_dir_file_count = build.get_file_count(
+            build.EXTENSIONS_DIRNAMES_TO_DIRPATHS['dev_dir'])
+        target_dir_file_count = build.get_file_count(
+            build.TEMPLATES_CORE_DIRNAMES_TO_DIRPATHS['dev_dir'])
+        self.assertTrue(
+            ('%s files in source dir != %s files in target dir.') % (
+                source_dir_file_count, target_dir_file_count) in
+            incorrectFileCount.exception)
+
+    def test_match_filename_with_hashes(self):
+        """Test _match_filename_with_hashes to raise exception:
+            1) When there is an empty hash dict.
+            2) When a filename is expected to contain hash but does not.
+            3) When there is a hash in filename that cannot be found in
+                hash dict.
+        """
+        # Final filepath example: base.240933e7564bd72a4dde42ee23260c5f.html.
+        file_hashes = dict()
+        base_filename = 'base.html'
+        with self.assertRaises(ValueError) as emptyHashDict:
```

Comment:
self.assertRaisesRegexp, here and below.

Btw, for future reference, everything in Python should be snake case (like empty_hash_dict). The exceptions with assertRaises etc. arise due to functions in unittest being defined differently from the usual Python norm.

Reply:
Made all variable snake case.